### PR TITLE
Added feature to subscribe to EvalAI on Home Page

### PR DESCRIPTION
**Pull request by jayaike**
#### Changes proposed in this pull request:

-  Added subscribe form to Home Page

#### DEMO VIDEO
bandicam-2019-12-04-12-45-41-252

### CODE REVIEW

**Comment by Sanji515** [comment]
@nsjcorps Please fix the travis build

**Comment by jayaike** [comment]
I have made the necessary changes to pass the lint. Sorry about that,

**Review by Ram81 — CHANGES_REQUESTED** [review]
@nsjcorps please change the PR name based on instructions

**Comment by jayaike** [comment]
I have done that sir.

**Comment by krtkvrm** [comment]
Hi @nsjcorps 
The button does work, but when I enter email id which was to already subscribed earlier, the error message `Something went wrong <400>` is popped up.
Screenshot 2019-12-06 at 6 25 02 PM

It would be better to display the message that clearly give the information to the user that the email is already subscribed and not a generic error. Something like, `You have already subscribed to EvalAI`

**Comment by jayaike** [comment]
Alright I will fix it to work as it should

**Comment by krtkvrm** [comment]
@nsjcorps 
I am getting following message when trying to subscribe with same email id:
Screenshot 2019-12-06 at 8 09 07 PM

**Comment by jayaike** [comment]
It should work well now. I was apparently checking the wrong property.

File: src/app/components/home/home.component.ts [hunk 1]
```diff
@@ -14,12 +15,23 @@ import {Router} from '@angular/router';
...
 })
-export class HomeComponent implements OnInit, OnDestroy {
-
+export class HomeComponent implements OnInit, AfterViewInit, OnDestroy {
   challengeCreateRoute = '/challenge-create';
   authRoute = '/auth/login';
   public user = {};
   public challengeList = [];
+
+  /**
```
Comment (Ram81) [inline src/app/components/home/home.component.ts]: realign this

File: src/app/components/home/home.component.ts [hunk 2]
```diff
@@ -56,9 +75,44 @@ export class HomeComponent implements OnInit, OnDestroy {
     }
   }
 
+  /**
+   * Form submit function (Called after validation).
+   * @param self  context value of this
```
Comment (Ram81) [inline src/app/components/home/home.component.ts]: remove extra space

File: src/app/components/home/home.component.ts [hunk 3]
```diff
@@ -56,9 +75,44 @@ export class HomeComponent implements OnInit, OnDestroy {
...
+    });
+    self.apiService.postUrl(self.endpointService.subscribeURL(), SUBSCRIBE_BODY).subscribe(
+      response => {
+        // Success Message in response.message
+        setTimeout(() => self.globalService.showToast('success', response.message, 5), 1000);
+      },
+      // Show Form Error on Failure
+      err => {
+        if (err.error.message) {
+          setTimeout(() => self.globalService.showToast('success', err.error.message, 5), 1000);
```
Comment (Ram81) [inline src/app/components/home/home.component.ts]: @nsjcorps why are we showing a `success` message when there is a error?

File: src/app/services/endpoints.service.ts [hunk 4]
```diff
@@ -75,6 +75,14 @@ export class EndpointsService {
   featuredChallengesURL() {
     return `${this.challenges}featured/`;
   }
+  /**
```
Comment (Ram81) [inline src/app/services/endpoints.service.ts:79]: Add a new line above this

File: src/app/services/endpoints.service.ts [hunk 5]
```diff
@@ -75,6 +75,14 @@ export class EndpointsService {
...
+   * Subscribe to Newsletters
+   */
+
```
Comment (Ram81) [inline src/app/services/endpoints.service.ts]: Remove new line

**Review by Ram81 — CHANGES_REQUESTED** [review]

**Comment by jayaike** [comment]
I have made the requested changes

File: src/app/components/home/home.component.ts [hunk 6]
```diff
@@ -56,9 +75,44 @@ export class HomeComponent implements OnInit, OnDestroy {
...
+    );
+  }
+
   ngOnDestroy(): void {
     if (this.authServiceSubscription) {
       this.authServiceSubscription.unsubscribe();
     }
   }
+
+
```
Comment (Ram81) [inline src/app/components/home/home.component.ts]: remove extra newline

File: src/app/components/home/home.component.ts [hunk 7]
```diff
@@ -14,12 +15,23 @@ import {Router} from '@angular/router';
   templateUrl: './home.component.html',
   styleUrls: ['./home.component.scss']
 })
-export class HomeComponent implements OnInit, OnDestroy {
-
+export class HomeComponent implements OnInit, AfterViewInit, OnDestroy {
```
Comment (Ram81) [inline src/app/components/home/home.component.ts:19]: Add a newline above this

**Review by Ram81 — CHANGES_REQUESTED** [review]
@nsjcorps some minor changes

**Review by RishabhJain2018 — APPROVED** [review]
LGTM :) Thanks @nsjcorps for adding it.
Thanks @Ram81 @vkartik97 @Sanji515 for the reviews.

**Comment by jayaike** [comment]
Thank you @RishabhJain2018 for merging the pull request and thank you @Sanji515 @vkartik97 @Ram81 for all the reviews.

**Pull request merged**